### DISCUSSION TITLE
noob guide to the config.ini file

### DISCUSSION THREAD

**Discussion by my-machines** [reinhard-brandstaedter/solarflow-control#137] [Q&A]
Hi everyone,

I am very glad  Solarflow Control exists, and I'm trying to set it up and integrate my smart meter right now, but I'm encountering some issues. I'd appreciate any assistance you can offer.

My questions:

- In which directory can I edit or create the config.ini file? I'm unsure about the appropriate location.

- How can I integrate my smart meter data from Home Assistant (specifically, the Frient Powermeter)? I'm not sure if Solarflow Control directly supports this integration, or what parameters are needed in the config.ini file (and where to find them in HA). 

- Additionally, what is meant by "rapid_change_diff"?

Thank you all very much in advance and please be patient with a noob trying to "decloud" himself and his home.

**Comment by mavo** [comment]
Hey,

> In which directory can I edit or create the config.ini file? I'm unsure about the appropriate location.

The config.ini is mounted into the docker container - and based on where you tell the container to read the config from (can't tell where it is on your machine) it takes it.

> How can I integrate my smart meter data from Home Assistant (specifically, the Frient Powermeter)? I'm not sure if Solarflow Control directly supports this integration, or what parameters are needed in the config.ini file (and where to find them in HA).

I hope that the maintainer of this repo jumps in and corrects me, but afaik its just not possible. You need to have one of the data sources that are supported, such as shellyem3, tasmota energy reader, etc.

> Additionally, what is meant by "rapid_change_diff"?

Imagine your home power consumption jumps (because you turn on the microwave) it'll create a spike in consumption. As the tool originally was build to have slower increase/decrease in power delivery it could happen that when the spike was over, you still drain the battery as the moving avg was calculated for a longer period and therefore still thinks your home needs more power than it actually does. The rapid change, as far as I understood it helps to reset the moving avg to be calculated again when the power changes more then the configured value.

MaVo

**Reply by my-machines** [reply]
Hey MaVo,

thank you very much for your elaborate answer, it is very much appreciated. Will try my luck again tonight.

mM
